Assets/Expansion2/CityBanners/citybannermanager.lua

```diff
@@ -4234,7 +4237,11 @@ function CQUI_RealHousingFromImprovements(pCity, PlayerID, pCityID)
           if not kPlot:IsImprovementPillaged() then
             local kImprovementData = GameInfo.Improvements[eImprovementType].Housing;
             if kImprovementData == 1 then    -- farms, pastures etc.
-              CQUI_HousingFromImprovements = CQUI_HousingFromImprovements + 1;
+              if GameInfo.Improvements[eImprovementType].ImprovementType == "IMPROVEMENT_FARM" and HasTrait("TRAIT_CIVILIZATION_MAYAB", Game.GetLocalPlayer()) then
+                CQUI_HousingFromImprovements = CQUI_HousingFromImprovements + 2;
+              else
+                CQUI_HousingFromImprovements = CQUI_HousingFromImprovements + 1;
+              end
```

Comment:
this can be refactored to "add 1, and then a separate corner case - if farm+trait, then add another one". no need in if-else here.

Reply:
it can be, yeah... I want to look into a larger scale refactor so I'll leave it this way for the time being